biz.aQute.bndlib/test/aQute/bnd/build/JarTest.java

```diff
@@ -0,0 +1,160 @@
+package aQute.bnd.build;
+
+import static org.junit.Assert.assertNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import aQute.bnd.classfile.ClassFile;
+import aQute.bnd.classfile.builder.ModuleInfoBuilder;
+import aQute.bnd.osgi.Constants;
+import aQute.bnd.osgi.EmbeddedResource;
+import aQute.bnd.osgi.Jar;
+import aQute.bnd.osgi.ManifestResource;
+import aQute.bnd.osgi.Resource;
+import aQute.lib.io.ByteBufferDataOutput;
+
+public class JarTest {
+
+	private static final String	MODULE_INFO_PATH			= "META-INF/versions/9/" + Constants.MODULE_INFO_CLASS;
+
+	private static final String	MAIN_MANIFEST_PATH			= JarFile.MANIFEST_NAME;
+
+	private static final String	SUPPLEMENTAL_MANIFEST_PATH	= "META-INF/versions/9/" + MAIN_MANIFEST_PATH;
+
```

Comment:
I think this is the wrong path as per the OSGi core spec (link in earlier comments).

Reply:
Fixed.